### DISCUSSION TITLE
Logs Explorer

### DISCUSSION THREAD

**Discussion by JulianSwales** [supabase/supabase#14952] [Questions]
I have a few questions about Logs Explorer. 
I may be missing something, but I can create a saved query (yay), however I don't see any way to update or delete Queries? So I now have multiple versions of the same query saved as i was iterating over getting it correct. 
We can run a saved as query, but I need to be able to pass in a value on the run button press. 
There seems to be no way to run these queries externally? 
Here's the scenario we have:
We are using email Otp for our customers. This works great, up until the customer has a link checker in their email process (which "clicks" the link prior to the user doing so, thereby invalidating it). Then we have customers complaining that their link is invalid and what are we doing too fix it. 
I have a query I can run in the Logs Explorer, that identifies the two IPS and times in this process, such that we can send an email to the customer with the information from the log and information from our email provider logs that they can send to their support people to have an exemption put in place. 
However, this is all manual currently on the supabase side. I want to put up an admin panel with a query to do this for our admin via their dashboard, but currently I don't see a way to do this unless I'm missing something? 

Thanks, Julian.

**Comment by GaryAustin1** [comment]
I don't know have answer on the log stuff, but have you considered the approach here https://supabase.com/docs/guides/auth/auth-email-templates#email-prefetching to avoid the email issue?

**Reply by JulianSwales** [reply]
Thanks Gary for pointing that out. I hadn't seen it. It's an interesting thought that we are thinking about..  We've managed to get most users working, by pulling the logs and providing the information for them to update on their end. It's a few users, hence us not rushing to change the whole process. 
Still would like the Log Access tho... It's very useful and handy for monitoring what's going on.

**Comment by Ziinc** [comment]
There are no plans to provide parameter interpolation in the studio, or to allow saved query editing. We also are not planning to provide programmatic access yet.

For your admin panel integration, I suggest building the sql query parameter using `?q=select...` which will execute the provided sql on page load.
